core-web/libs/data-access/src/lib/ema-app-configuration/ema-app-configuration.service.ts

```diff
@@ -0,0 +1,112 @@
+import { EMPTY, Observable } from 'rxjs';
+
+import { HttpClient } from '@angular/common/http';
+import { Injectable, inject } from '@angular/core';
+import { Router } from '@angular/router';
+
+import { pluck, switchMap, map, defaultIfEmpty, catchError } from 'rxjs/operators';
+
+import { SiteService } from '@dotcms/dotcms-js';
+import { DotAppsSite, DotApp } from '@dotcms/dotcms-models';
+
+import { DotLicenseService } from '../dot-license/dot-license.service';
+
+interface EmaAppSecretValue {
+    pattern: string;
+    url: string;
+    options: EmaAppOptions;
+}
+
+interface EmaAppOptions {
+    authenticationToken: string;
+    [key: string]: string;
+}
+
+@Injectable()
+export class EmaAppConfigurationService {
+    http = inject(HttpClient);
+    router = inject(Router);
+    licenseService = inject(DotLicenseService);
+    siteService = inject(SiteService);
+
+    /**
+     * Get the EMA app configuration for the current site.
+     *
+     * @param {string} url
+     * @return {*}  {(Observable<EmaAppSecretValue | null>)}
+     * @memberof EmaAppConfigurationService
+     */
+    get(url: string): Observable<EmaAppSecretValue | null> {
+        return this.licenseService
+            .isEnterprise()
+            .pipe(
+                map((isEnterprise) => {
+                    if (!isEnterprise) {
+                        throw new Error('Not enterprise');
+                    }
+
+                    return isEnterprise;
+                })
+            )
+            .pipe(
+                switchMap(() => this.getCurrentSiteIdentifier()),
+                switchMap((currentSiteId: string) => {
+                    return this.getEmaAppConfiguration(currentSiteId).pipe(
+                        map(getConfigurationForCurrentSite(currentSiteId))
+                    );
+                }),
+                map(getSecretByUrlMatch(url)),
+                catchError(() => {
+                    return EMPTY;
+                }),
+                defaultIfEmpty<EmaAppSecretValue | null>(null)
+            );
+    }
+
+    private getCurrentSiteIdentifier(): Observable<string> {
+        return this.siteService.getCurrentSite().pipe(pluck('identifier'));
+    }
+
+    private getEmaAppConfiguration(id: string): Observable<DotApp> {
+        return this.http
+            .get<{ entity: DotApp }>(`/api/v1/apps/dotema-config-v2/${id}`)
+            .pipe(pluck('entity'));
+    }
+}
+
+function getConfigurationForCurrentSite(currentSiteId: string): (app: DotApp) => DotAppsSite {
```

Comment:
We should consider moving this to a `util.ts`

Reply:
I don't see we're using this anywhere else, this service is very specific, if we change the arq of the service then we can see.